### ISSUE TITLE
hackage-root-tool create-key fails on Windows

### ISSUE THREAD

**Issue by snoyberg** [well-typed/hackage-security#115]
Sample output:

```
E:\>hackage-root-tool.exe create-key
Writing new key:
  7dbb1f42d0af469848dca2de61214f218afd3e5fcc7288c5333739c9e4ba5309.{private,public}
fromAbsoluteFilePath: not an absolute path
```

I could generate the root key in my Linux VM instead, but that would mean the key is traversing *two* systems instead of being generated directly on my USB drive, which is what I'm trying to avoid.

**Comment by hvr** [comment]
I'm sure this is considered a bug :-)

**Comment by snoyberg** [comment]
Oh, it's actually easy to discover where this bug comes from:

```haskell
fromAbsoluteFilePath ('/':path) = rootPath Rooted (fromUnrootedFilePath path)
fromAbsoluteFilePath _ = error "fromAbsoluteFilePath: not an absolute path"
```

That should be using functions from System.FilePath instead. I'll send a PR for that, but it still leaves me in a difficult position for rebuilding the tool, given that the version of hackage-security in the repo is blocked by upper bounds in the hackage-root-tool.cabal file.

Is the upper bound intentional? I'd recommend adding a Travis job that confirms all subdirs compile together correctly. I can send you a PR for doing that based on stack's Travis scripts if you like.

**Comment by hvr** [comment]
re Travis-CI, this reminds me I already offered @edsko to setup a Travis-CI job in the style of https://github.com/haskell/test-framework/blob/master/.travis.yml (which also features caching). I just forgot to followup... :-(

**Comment by snoyberg** [comment]
Perhaps I spoke too soon about the easy-to-fix error: it seems like the `Path` datatype itself is fairly hard-coded with Unix specific paths, disallowing Windows ones. I'm actually not sure how to proceed with fixing that.

**Comment by snoyberg** [comment]
I'll send the Travis PR, I think 90% of my day job is configuring CI systems ;)

**Comment by hvr** [comment]
> it seems like the Path datatype itself is fairly hard-coded with Unix specific paths, disallowing Windows ones.

heh... this reminds me about another thing I forgot to followup... AFPP ;-)

**Comment by snoyberg** [comment]
PR #116 sent

**Comment by edsko** [comment]
Opened a separate ticket about the path issue.

**Comment by edsko** [comment]
With the `Path` module fixed this works on Windows for me now, building with `stack`, provided I set the `old-time` flag of `tar` to `False` (see https://github.com/haskell/tar/pull/11).
